jdaviz/configs/default/plugins/viewers.py

```diff
@@ -124,6 +124,8 @@ def _on_layers_update(self, layers=None):
             return
 
         viewer_item = self.jdaviz_app._viewer_item_by_id(self.reference_id)
+        if viewer_item is None:
+            return
```

Comment:
this gets triggered when the viewer is removed from the app before the layer update messages are processed?  If so, I think this makes sense as a good way to quickly prevent that case from causing issues.  We don't need to worry about any of the following logic (which only affects things in the viewer, not higher in the stack) since the viewer is no longer in the app.

Reply:
This one is for #1614 . I don't grok all the callback stuff but when this happens:

```
  114 viewer_item = self.jdaviz_app._viewer_item_by_id(self.reference_id)
 --> 115 selected_data_items = viewer_item['selected_data_items']
```

`self.reference_id` actually points to the deleted viewer ID (not sure why, maybe upstream bug). So I think returning here is good enough until we find a better fix.